src/filters/match_pattern.rs

```diff
@@ -0,0 +1,188 @@
+use gasket::framework::*;
+use pallas::{
+    ledger::addresses::{Address, StakeAddress},
+    network::miniprotocols::Point,
+};
+use serde::Deserialize;
+use tracing::error;
+
+use crate::framework::*;
+
+#[derive(Stage)]
+#[stage(name = "filter-match-pattern", unit = "ChainEvent", worker = "Worker")]
+pub struct Stage {
+    predicate: Predicate,
+
+    pub input: FilterInputPort,
+    pub output: FilterOutputPort,
+
+    #[metric]
+    ops_count: gasket::metrics::Counter,
+}
+
+#[derive(Default)]
+pub struct Worker;
+
+impl From<&Stage> for Worker {
+    fn from(_: &Stage) -> Self {
+        Worker::default()
+    }
+}
+
+gasket::impl_splitter!(|_worker: Worker, stage: Stage, unit: ChainEvent| => {
```

Comment:
I believe that `impl_mapper` is better suited for this case.

Reply:
impl_mapper doesn't implement return Option